gradle/java.gradle

```diff
@@ -12,6 +12,7 @@ compileJava {
 
 task testJar(type: Jar, dependsOn: testClasses, group: "build") {
     appendix = 'test'
+    classifier = 'tests'
```

Comment:
@mmalohlava Turns out I was wrong, it does change the name of the jar to h2o-core-test-3.13.0-tests.jar

It was only working because I had also other changes in my build. Fixed now in next commit.

Please review again.